app/models/metrics.go

```diff
@@ -303,6 +297,7 @@ type specialMetrics struct {
 type rateMetrics struct {
 	Point                            int64
 	Ts                               time.Time
+	NoData                           bool
```

Comment:
Maybe double negation is not the best e.g. I would prefer `Data=true` instead of `NoData=false` - just suggestion for the future.

Reply:
Yeap, good suggestion. But for now, it is same as message in sparkling.